client/src/components/ElementActions/ArchiveAction.js

```diff
@@ -0,0 +1,46 @@
+/* global confirm */
+import React from 'react';
+import { compose } from 'redux';
+import AbstractAction from 'components/ElementActions/AbstractAction';
+import archiveBlockMutation from 'state/editor/archiveBlockMutation';
+import i18n from 'i18n';
+
+/**
+ * Adds the elemental menu action to archive a block of any state
+ */
+const ArchiveAction = (WrappedComponent) => (props) => (
+  <WrappedComponent {...props}>
+    {props.children}
+
+    <AbstractAction
+      title={i18n._t('ArchiveAction.ARCHIVE', 'Archive')}
+      extraClass="element-editor__actions-archive"
+      onClick={(event) => {
```

Comment:
How about:

```js
ArchiveAction = (Thing) => (props) => {
	const handleClick = () => {...}
	return (
		...
		onClick={handleClick}
		...
	)
}
```
?

Reply:
Really this doesn't make a difference, I was just suggesting for readability. Not really fussed here.

Reply:
Sure